vault/resource_pki_secret_backend_config_est.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/util/mountutil"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func pkiSecretBackendConfigEstResource() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manages Vault PKI EST configuration",
+		CreateContext: provider.MountCreateContextWrapper(pkiSecretBackendConfigEstWrite, provider.VaultVersion116),
+		UpdateContext: pkiSecretBackendConfigEstWrite,
+		ReadContext:   pkiSecretBackendConfigEstRead,
+		DeleteContext: pkiSecretBackendConfigEstDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(_ context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				id := d.Id()
```

Comment:
My understanding is the main reason for adding this bit is that both the data source and the resource are sharing the same `readPKISecretBackendConfigEst` method. While I see the benefits of having a common util, I think splitting them out a little might be beneficial for us in the long run. In my experience, it's always better to try and avoid writing custom importers as much as possible. They sometimes behave unexpectedly in edge cases where users input weird things 😅 

Once things get garbled in the TF state for users in production, it's hard for us to rollback these changes (especially with a field like `backend`, which is a `ForceNew` and would attempt to destroy and recreate)

Reply:
My suggestion would be:
- For the data source, verify the Vault server version requirements. We also can always expect that the `backend` field will be provided (`d.GetOk` is allowed)
- For the resource, we don't need to verify the Vault server version requirements again (already taken care of by `MountCreateContextWrapper`). For the read method, we should extract the `backend` parameter from the `d.Id()` instead. I would suggest stripping the end `/config/est` instead of splitting the string on a `/`. We can never expect how many slashes users might put into their input values, so this can lead to incorrect TF states

Typically for Read methods in resources, we try not to have any `d.Get` commands in order to make the import work seamlessly. We should try to construct our IDs and resources in a way where all relevant info for the TF state is available through the read method.

Let me know if this makes sense! Happy to chat about this further if you had other thoughts :)

Reply:
I think I understand have a look at https://github.com/hashicorp/terraform-provider-vault/pull/2246/commits/8183fe41f06467d6c25bb1f4b6484bbfeff8bf2e, I'm assuming we still need to keep the requirements check due to the enterprise requirement.